wgengine/magicsock/magicsock.go

```diff
@@ -511,21 +511,33 @@ func appendDests(dsts []*net.UDPAddr, as *AddrSet, b []byte) (_ []*net.UDPAddr,
 
 	// Pick our destination address(es).
 	roamAddr = as.roamAddr
-	if roamAddr != nil {
-		dsts = append(dsts, roamAddr)
-		if !spray {
-			return dsts, roamAddr
+	switch {
+	case spray:
+		// This packet is being sprayed to all addresses.
+		for i := range as.addrs {
+			dsts = append(dsts, &as.addrs[i])
 		}
-	}
-	for i := len(as.addrs) - 1; i >= 0; i-- {
-		addr := &as.addrs[i]
-		if spray || as.curAddr == -1 || as.curAddr == i {
-			dsts = append(dsts, addr)
+		if as.roamAddr != nil {
+			dsts = append(dsts, as.roamAddr)
 		}
-		if !spray && len(dsts) != 0 {
-			break
+	case as.roamAddr != nil:
+		// We have a roaming address, prefer it over other addrs.
+		// TODO(danderson): this is not correct, there's no reason
+		// roamAddr should be special like this.
+		dsts = append(dsts, as.roamAddr)
+	case as.curAddr != -1 && as.curAddr < len(as.addrs):
```

Comment:
The second half of this condition should always be true. Please invert it and move it inside the case and log when it happens.